packages/router/src/router.js

```diff
@@ -115,12 +115,17 @@ const RouterImpl = ({
         .map((privateElement) => {
           // Set `Route` props
           const { unauthenticated, role, children } = privateElement.props
-          return React.Children.toArray(children).map((route) =>
-            React.cloneElement(route, {
-              private: true,
-              unauthenticatedRedirect: unauthenticated,
-              role: role,
-            })
+          return (
+            React.Children.toArray(children)
+              // Make sure only valid objects are considered
+              .filter((x) => typeof x === 'object')
```

Comment:
I think you could do something like we're doing above on line :114

`.filter((child) => child.type === Route)`

Reply:
thanks for pointing that out! I thought it looked weird but didn't figure out how to make it better.
It would have to be `.filter((route) => route.type === Route)` though
updating PR in a sec

Reply:
Thanks!